### ISSUE TITLE
Unable to create the server: Extension backend error - failed to download runtime in time

### ISSUE THREAD

**Issue by Intradiction** [redhat-developer/vscode-server-connector#526]
I receive that error when trying to download Tomcat servers, is it possibly caused by an incompatible Java version?

**Comment by andreanaFelicione** [comment]
I have the same error.

**Comment by nicocossiom** [comment]
Same

**Comment by robstryker** [comment]
Are you all using java 17?

I'll try to get a release out ASAP with java 17 support. The new extension should be backwards compatible with previous java versions too, if everything goes as expected.

**Comment by nicocossiom** [comment]
I have tried using both openjdk11 and 17 and they both don't work. What doesn't work is downloading a server through the extension. Although I can download the server binaries separately and then use it normally.
Trying to download a server through the extension.

https://user-images.githubusercontent.com/72339792/161377249-b3bf80fb-6277-40d6-bdbb-fddd1f200ad9.mp4

Using downlaoded binaries:


https://user-images.githubusercontent.com/72339792/161377257-3ca66403-2f3c-466a-bada-6f7b05c11e2a.mp4

**Comment by Joohangooi** [comment]
Are you guys using M1 mac as well? I have tried to create a server on a Windows pc, it works but not on my M1 Mac.

**Comment by nicocossiom** [comment]
> Are you guys using M1 mac as well? I have tried to create a server on a Windows pc, it works but not on my M1 Mac. 

Yeah M1 on MBPro 2020. But i have the same problem on my Ryzen desktop

**Comment by spaiq** [comment]
Got the same problem while trying to download the server, but when I try to use the one from disk I get an
"Error: Could not find or load main class Files
Caused by: java.lang.ClassNotFoundException: Files" error.
Does anybody know how to solve at least this one?

**Comment by robstryker** [comment]
Is this issue still replicatable with newer runtimes? There was an issue in the past where some download urls changed.

**Comment by lloydie-t** [comment]
Same Problem on windows 11 (64 Bit), JDK17, Paraya 6.2022.2 and Wildfly 27.0.1 final. The packaged ones install fine, but do not seem to ne able to deploy my war files.

Unable to create the server:  Extension backend error - could not detect any server at c:\users\someuser\downloads\wildfly-27.0.1.final\wildfly-27.0.1.final!

**Comment by Terry-fengweilun** [comment]
I have a solution to the problem
mine: Mac m1 MacBook-Air vscode tomcat server
1.Finder->/users/ cmd+shift+. (display hidden files)-> /Users/***/.rsp/redhat-community-server-connector/runtimes/installations/ 
2.here paste your own Tomcat downloaded from the official website.
3.finally, in Vscode,reload the new Tomcat address.
you will find the PERMISSION turns to admitted and it can load normally.

**Comment by robstryker** [comment]
It's really strange. We don't expect the download to complete quickly, but we do expect the RSP server to respond that a download has been initiated within 7.5 seconds. Then there should be job / progress report on the download.

**Comment by Elizura** [comment]
Has this issue been fixed?
I am using kali linux and the error still persists

**Comment by Matthew7712** [comment]
I used this guide to solve this problem : https://code.visualstudio.com/docs/java/java-tomcat-jetty.
First, I deleted the server.
![Screenshot from 2024-07-24 00-33-48](https://github.com/user-attachments/assets/ecc40d96-0b31-43e5-9eb3-1f275854fd8c)
Then followed the steps outlined in the user manual.
1. Add new Apache Tomcat Server;
2. Start the created server;
3. Right-click on the new server, select Server Actions... from the menu, then select Show in Browser..., then select http://localhost:8080/.;
4. And last step, return to vs code, then find war archive of your app, and right click on it. This will bring up a menu, in it select Run on Server.

**Comment by robstryker** [comment]
I was able to replicate this and fixed it here:   https://github.com/redhat-developer/rsp-server/issues/740

It was due to a stalled download of the license. The video confirms it as well. Download runtime, yes, select a type, and the license never appears. It was stalled waiting for the license to download. 

Fixed as per the above issue.
